pkg/storage/tsdb/config.go

```diff
@@ -456,15 +460,23 @@ func (cfg *BucketStoreConfig) Validate(logger log.Logger) error {
 }
 
 type BucketIndexConfig struct {
-	Enabled               bool          `yaml:"enabled"`
+	DeprecatedEnabled     bool          `yaml:"enabled" category:"deprecated"` // Deprecated. TODO: Remove in Mimir 2.11.
 	UpdateOnErrorInterval time.Duration `yaml:"update_on_error_interval" category:"advanced"`
 	IdleTimeout           time.Duration `yaml:"idle_timeout" category:"advanced"`
 	MaxStalePeriod        time.Duration `yaml:"max_stale_period" category:"advanced"`
 }
 
 func (cfg *BucketIndexConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
-	f.BoolVar(&cfg.Enabled, prefix+"enabled", true, "If enabled, queriers and store-gateways discover blocks by reading a bucket index (created and updated by the compactor) instead of periodically scanning the bucket.")
+	f.BoolVar(&cfg.DeprecatedEnabled, prefix+"enabled", true, "If enabled, queriers and store-gateways discover blocks by reading a bucket index (created and updated by the compactor) instead of periodically scanning the bucket.")
```

Comment:
Since we no longer care about this setting and will always treat it as `true` in the future, should we use `flagext.DeprecatedFlag` and remove all the logic that depends on it?